[Response Ops][Task Manager] Allow schedule updates when `ensureScheduled` API is used

ymao1 opened the pull request:
Resolves https://github.com/elastic/kibana/issues/222089

## Summary

Updates `ensureScheduled` API to update task schedule if task already exists using the existing `bulkUpdateSchedules` method. Currently `bulkUpdateSchedules` only handle schedule intervals (not rrule schedules) so this only works for intervals, which I believe is ok because rrule is not widely used yet.

ymao1 committed:
Allow schedule updates when ensureScheduled is used

elena-shostak committed:
[Spaces] Deleted obsolete test for session expiration toast (#237876)

## Summary

In scope of https://github.com/elastic/kibana/pull/235957 session
timeout toast with `Extend` button has been moved to a modal. Deleted
obsolete test that is no longer valid.


### Checklist

- [x] The PR description includes the appropriate Release Notes section,
and the correct `release_note:*` label is applied per the
[guidelines](https://www.elastic.co/guide/en/kibana/master/contributing.html#kibana-release-notes-process)
- [x] Review the [backport
guidelines](https://docs.google.com/document/d/1VyN5k91e5OVumlc0Gb9RPa3h1ewuPE705nRtioPiTvY/edit?usp=sharing)
and apply applicable `backport:*` labels.

__Related: https://github.com/elastic/kibana/issues/138333__

SrdjanLL committed:
[kbn-evals] Assign meaningful experiment and task names for Agent Builder evals (#237892)

## Summary

- Assign experiment name in the format of `Run ID: <test_run_id>
Dataset: <phoenix dataset name>`.
- Will make it easier to understand which dataset experiment is running
against in the terminal
- The run id is common for all evaluations in the same run. It will be
used for exporting results to Elasticsearch and building reports.
- Providing a task as a named function in Agent Builder's
`createEvaluateDataset` assigns the function name as the task name in
the Phoenix experiment, providing more meaningful logs.


## Testing
- Ran Agent Builder evals and confirmed the logs use the new names:
```bash
 info [scout-worker] 🔗 View this experiment: http://localhost:6006/datasets/RGF0YXNldDo0/compare?experimentId=RXhwZXJpbWVudDoxMTE=
 info [scout-worker] 🧪 Starting experiment "Run ID: 874bf4bdbf34a60d - Dataset: onechat: default-agent-unanswerable-queries" on dataset "RGF0YXNldDo0" with task "callConverseAndEvaluate" and 4 evaluators and 5 concurrent runs
 info [scout-worker] 🔧 Running task "callConverseAndEvaluate" on dataset "RGF0YXNldDo0"
 info [scout-worker] 🔧 Running task "callConverseAndEvaluate" on example "RGF0YXNldEV4YW1wbGU6NTQ= of dataset "RGF0YXNldDo0"
```
- New naming makes comparisons within Phoenix UI a bit more intuitive
since names use the test run IDs. The same run IDs will be used for
reporting (see
[here](https://github.com/elastic/kibana/blob/main/x-pack/platform/packages/shared/kbn-evals/src/utils/score_repository.ts#L285))
<img width="642" height="233" alt="image"
src="https://github.com/user-attachments/assets/1bfdd555-1103-45af-b3d8-7b87c08e29a0"
/>

ymao1 commented on x-pack/platform/plugins/shared/task_manager/server/task_scheduling.ts:
```diff
@@ -284,12 +284,23 @@ export class TaskScheduling {
+        // if so,try to update the just the schedule
+        // only works for interval schedule
+        if (taskInstance.schedule && taskInstance.schedule.interval) {
+          const result = await this.bulkUpdateSchedules([taskInstance.id], taskInstance.schedule);
```
`bulkUpdateSchedules` already filters out tasks with no change to the schedule interval so no check needed here.

simianhacker committed:
[Metrics][Discover] Refactor chart layers hook (#237390)

## 🍒 Summary

This change refactors the `useChartLayers` hook in the Unified Metrics
Grid to generate the layers based on the metric fields instead of the
ESQL query. The previous `useChartLayers` hook created a request with
`LIMIT 0` to get the columns for the chart to build the layers, this
created an unnecessary network request for each chart.

The change also introduces a ES|QL parameter substitution utility to
ensure the columns defined in the chart layers matches the output from
the ESQL query. It also introduces a utility to generate the metric
aggregation and time bucket aggregation uses for both the chart layers
and ESQL creation utility to ensure they match.

## 🛠️ Changes

- Created a comprehensive unit test suite for the new `useChartLayers`
hook to ensure its correctness.
- Updated the `Chart` component and `useLensProps` hook to consume the
new `useChartLayers` hook, simplifying their implementation.
- Added a new `create_aggregation.ts` utility with a
`replaceFunctionParams` function to handle ES|QL parameter substitution
and escaping.
- Included a unit test for the new aggregation helpers to verify their
behavior.
- Added JSDoc documentation blocks to `createMetricAggregation` and
`createTimeBucketAggregation` functions for better code clarity.

## 🎙️ Prompts

- "Can you create a function like `escapeString("system.load.1m")` that
would return "system.load.`1m`" using the same technique."
- "can you add that to
@src/platform/packages/shared/kbn-unified-metrics-grid/src/common/utils/esql/create_aggregation.ts"
- "Hum... Modify it to do this
`replaceFunctionParams('AVG(??metricField)', { metricField:
'system.load.1m' })` which would output `AVG(system.load.\`1m\`)`"
- "I think you miss understood what I want. I want a function that I can
call with `AVG(??metricField)` and `{ metricField: 'system.load.1m' }`
and it will do the substitution using the AST."
- "Params and replaceParameters are not exported from the
`esql-composer` library. Can you read
@src/platform/packages/shared/kbn-esql-composer/README.md Could use this
library to basically do the same thing?"
- "Can you write a unit test for this?"
- "I feel like what you originally wrote should work. It's working
pretty much the same way here:
@src/platform/packages/shared/kbn-unified-metrics-grid/src/common/utils/esql/create_esql_query.ts"
- "Can you add doc blocks to the other functions in the
create_aggregation file?"
- "Can you convert the `const chartLayers = useMemo...` in
@src/platform/packages/shared/kbn-unified-metrics-grid/src/components/chart/index.tsx
to a hook called `useChartLayers` and put it in
src/platform/packages/shared/kbn-unified-metrics-grid/src/components/chart/hooks/use_chart_layers.ts
and add a corresponding test?"

🤖 This pull request was assisted by Gemini CLI

ymao1 commented on x-pack/platform/plugins/shared/task_manager/server/task_scheduling.ts:
```diff
@@ -284,12 +284,23 @@ export class TaskScheduling {
       return await this.schedule(taskInstance, options);
     } catch (err) {
       if (err.statusCode === VERSION_CONFLICT_STATUS) {
+        // check if task specifies a schedule interval
```
I spent some time thinking about a way we could do this more generically (to allow for updates to more than just the schedule) but our task update functions are currently also very targeted (just enable/disable and update state and schedule) so to do a more would require a much larger change. I think we can wait to see if we get another request to update something during `ensureScheduled` to try to tackle a bigger refactor.

machadoum committed:
[Security Solution] Create a migration to auto-install integration sync entity source saved objects (#237406)

## Summary

Create a migration to auto-install integration sync entity source saved
objects.

* I also extracted the logger and audit logic from the data-client

### How to test it?
* Start kibana and start the PrivMon engine in a space
* Delete an entity source
* Get all sources `GET
kbn:/api/entity_analytics/monitoring/entity_source/list`
* Delete source by id `DELETE
kbn:/api/saved_objects/entity-analytics-monitoring-entity-source/{id}`
* Restart Kibana
* The source should have been recreated
* Get all sources `GET
kbn:/api/entity_analytics/monitoring/entity_source/list`

** Please test on multiple spaces
** Please test the upgrade scenario from 9.1 (with engine installed)

### Checklist

Check the PR satisfies following conditions. 

Reviewers should verify this PR satisfies this list as well.

- [ ] Any text added follows [EUI's writing
guidelines](https://elastic.github.io/eui/#/guidelines/writing), uses
sentence case text and includes [i18n
support](https://github.com/elastic/kibana/blob/main/src/platform/packages/shared/kbn-i18n/README.md)
- [ ]
[Documentation](https://www.elastic.co/guide/en/kibana/master/development-documentation.html)
was added for features that require explanation or tutorials
- [ ] [Unit or functional
tests](https://www.elastic.co/guide/en/kibana/master/development-tests.html)
were updated or added to match the most common scenarios
- [ ] If a plugin configuration key changed, check if it needs to be
allowlisted in the cloud and added to the [docker
list](https://github.com/elastic/kibana/blob/main/src/dev/build/tasks/os_packages/docker_generator/resources/base/bin/kibana-docker)
- [ ] This was checked for breaking HTTP API changes, and any breaking
changes have been approved by the breaking-change committee. The
`release_note:breaking` label should be applied in these situations.
- [ ] [Flaky Test
Runner](https://ci-stats.kibana.dev/trigger_flaky_test_runner/1) was
used on any tests changed
- [ ] The PR description includes the appropriate Release Notes section,
and the correct `release_note:*` label is applied per the
[guidelines](https://www.elastic.co/guide/en/kibana/master/contributing.html#kibana-release-notes-process)
- [ ] Review the [backport
guidelines](https://docs.google.com/document/d/1VyN5k91e5OVumlc0Gb9RPa3h1ewuPE705nRtioPiTvY/edit?usp=sharing)
and apply applicable `backport:*` labels.

### Identify risks

Does this PR introduce any risks? For example, consider risks like hard
to test bugs, performance regression, potential of data loss.

Describe the risk, its severity, and mitigation for each identified
risk. Invite stakeholders and evaluate how to proceed before merging.

- [ ] [See some risk
examples](https://github.com/elastic/kibana/blob/main/RISK_MATRIX.mdx)
- [ ] ...

---------

Co-authored-by: kibanamachine <42973632+kibanamachine@users.noreply.github.com>
Co-authored-by: Elastic Machine <elasticmachine@users.noreply.github.com>

rogercoll committed:
chore: bump kube-stack Helm chart onboarding to 0.10.5 (#237227)

## Summary

New kube-stack version that fixes Chart uninstallation for any version
prior 0.10.5:
https://github.com/open-telemetry/opentelemetry-helm-charts/issues/1876

 elastic-agent PR: https://github.com/elastic/elastic-agent/pull/10145

### Checklist

Check the PR satisfies following conditions. 

Reviewers should verify this PR satisfies this list as well.

- [ ] Any text added follows [EUI's writing
guidelines](https://elastic.github.io/eui/#/guidelines/writing), uses
sentence case text and includes [i18n
support](https://github.com/elastic/kibana/blob/main/src/platform/packages/shared/kbn-i18n/README.md)
- [ ]
[Documentation](https://www.elastic.co/guide/en/kibana/master/development-documentation.html)
was added for features that require explanation or tutorials
- [ ] [Unit or functional
tests](https://www.elastic.co/guide/en/kibana/master/development-tests.html)
were updated or added to match the most common scenarios
- [ ] If a plugin configuration key changed, check if it needs to be
allowlisted in the cloud and added to the [docker
list](https://github.com/elastic/kibana/blob/main/src/dev/build/tasks/os_packages/docker_generator/resources/base/bin/kibana-docker)
- [ ] This was checked for breaking HTTP API changes, and any breaking
changes have been approved by the breaking-change committee. The
`release_note:breaking` label should be applied in these situations.
- [ ] [Flaky Test
Runner](https://ci-stats.kibana.dev/trigger_flaky_test_runner/1) was
used on any tests changed
- [ ] The PR description includes the appropriate Release Notes section,
and the correct `release_note:*` label is applied per the
[guidelines](https://www.elastic.co/guide/en/kibana/master/contributing.html#kibana-release-notes-process)
- [ ] Review the [backport
guidelines](https://docs.google.com/document/d/1VyN5k91e5OVumlc0Gb9RPa3h1ewuPE705nRtioPiTvY/edit?usp=sharing)
and apply applicable `backport:*` labels.

### Identify risks

Does this PR introduce any risks? For example, consider risks like hard
to test bugs, performance regression, potential of data loss.

Describe the risk, its severity, and mitigation for each identified
risk. Invite stakeholders and evaluate how to proceed before merging.

- [ ] [See some risk
examples](https://github.com/elastic/kibana/blob/main/RISK_MATRIX.mdx)
- [ ] ...

---------

Co-authored-by: Joe Reuter <johannes.reuter@elastic.co>
Co-authored-by: kibanamachine <42973632+kibanamachine@users.noreply.github.com>

justinkambic committed:
[Synthetics] Do not create default alerts if connectors are not defined (#237504)

## Summary

Related to #221863. We wanted to merge this change regardless, even
though we have once again refined our approach to solving the problem of
default alerting in Synthetics.

mistic committed:
chore(NA): update versions after v9.0.9 bump (#237687)

This PR is a simple update of our versions file after the recent bumps.

---------

Co-authored-by: kibanamachine <42973632+kibanamachine@users.noreply.github.com>

mistic committed:
chore(NA): update versions after v9.1.6 bump (#237690)

This PR is a simple update of our versions file after the recent bumps.

---------

Co-authored-by: kibanamachine <42973632+kibanamachine@users.noreply.github.com>

mistic committed:
chore(NA): update versions after v8.18.9 bump (#237693)

This PR is a simple update of our versions file after the recent bumps.

---------

Co-authored-by: kibanamachine <42973632+kibanamachine@users.noreply.github.com>

mistic committed:
chore(NA): update versions after v8.19.6 bump (#237695)

This PR is a simple update of our versions file after the recent bumps.

---------

Co-authored-by: kibanamachine <42973632+kibanamachine@users.noreply.github.com>

jen-huang committed:
[UII] Add "Show agentless resources" toggle (#237528)

## Summary

Resolves https://github.com/elastic/ingest-dev/issues/5932.

This PR replaces the `?showAgentless` query param (implemented in
https://github.com/elastic/kibana/pull/219175) with a local storage
setting called `fleet:showAgentlessResources` that can be toggled from
Fleet > Settings. When enabled, agentless agents and policies are
visible in the Fleet UI:

![Oct-03-2025
11-56-35](https://github.com/user-attachments/assets/ebba0964-698c-4556-8798-2317fce938df)

## Release note
"Show agentless resources" toggle added to Fleet > Settings for
debugging and diagnostics purposes.

## To-do
- [ ] Public documentation:
https://github.com/elastic/ingest-docs/issues/1865

### Checklist

- [x] Any text added follows [EUI's writing
guidelines](https://elastic.github.io/eui/#/guidelines/writing), uses
sentence case text and includes [i18n
support](https://github.com/elastic/kibana/blob/main/src/platform/packages/shared/kbn-i18n/README.md)
- [ ]
[Documentation](https://www.elastic.co/guide/en/kibana/master/development-documentation.html)
was added for features that require explanation or tutorials
- [x] [Unit or functional
tests](https://www.elastic.co/guide/en/kibana/master/development-tests.html)
were updated or added to match the most common scenarios
- [x] The PR description includes the appropriate Release Notes section,
and the correct `release_note:*` label is applied per the
[guidelines](https://www.elastic.co/guide/en/kibana/master/contributing.html#kibana-release-notes-process)
- [x] Review the [backport
guidelines](https://docs.google.com/document/d/1VyN5k91e5OVumlc0Gb9RPa3h1ewuPE705nRtioPiTvY/edit?usp=sharing)
and apply applicable `backport:*` labels.

---------

Co-authored-by: Visha Angelova <91186315+vishaangelova@users.noreply.github.com>

VladimirFilonov committed:
Fix how testRun processing manual inputs (#237914)

## Summary

Bug https://github.com/elastic/security-team/issues/14174 was caused by
wrong processing of manual inputs. This PR aligns behaviour with actual
run logic


### Checklist

Check the PR satisfies following conditions. 

Reviewers should verify this PR satisfies this list as well.

- [ ] Any text added follows [EUI's writing
guidelines](https://elastic.github.io/eui/#/guidelines/writing), uses
sentence case text and includes [i18n
support](https://github.com/elastic/kibana/blob/main/src/platform/packages/shared/kbn-i18n/README.md)
- [ ]
[Documentation](https://www.elastic.co/guide/en/kibana/master/development-documentation.html)
was added for features that require explanation or tutorials
- [ ] [Unit or functional
tests](https://www.elastic.co/guide/en/kibana/master/development-tests.html)
were updated or added to match the most common scenarios
- [ ] If a plugin configuration key changed, check if it needs to be
allowlisted in the cloud and added to the [docker
list](https://github.com/elastic/kibana/blob/main/src/dev/build/tasks/os_packages/docker_generator/resources/base/bin/kibana-docker)
- [ ] This was checked for breaking HTTP API changes, and any breaking
changes have been approved by the breaking-change committee. The
`release_note:breaking` label should be applied in these situations.
- [ ] [Flaky Test
Runner](https://ci-stats.kibana.dev/trigger_flaky_test_runner/1) was
used on any tests changed
- [ ] The PR description includes the appropriate Release Notes section,
and the correct `release_note:*` label is applied per the
[guidelines](https://www.elastic.co/guide/en/kibana/master/contributing.html#kibana-release-notes-process)
- [ ] Review the [backport
guidelines](https://docs.google.com/document/d/1VyN5k91e5OVumlc0Gb9RPa3h1ewuPE705nRtioPiTvY/edit?usp=sharing)
and apply applicable `backport:*` labels.

### Identify risks

Does this PR introduce any risks? For example, consider risks like hard
to test bugs, performance regression, potential of data loss.

Describe the risk, its severity, and mitigation for each identified
risk. Invite stakeholders and evaluate how to proceed before merging.

- [ ] [See some risk
examples](https://github.com/elastic/kibana/blob/main/RISK_MATRIX.mdx)
- [ ] ...

machadoum committed:
[SecurirySolution] Fix risk score visu fails when entity name contains backslash (#237757)

## Summary

Add `"` around the entity name so Lens know that it has to escape
special characters.


<img width="3440" height="1340" alt="Screenshot 2025-10-07 at 08 19 44"
src="https://github.com/user-attachments/assets/6c0b6896-5529-430e-964b-afba1de37d95"
/>
<img width="1078" height="666" alt="Screenshot 2025-10-07 at 08 30 44"
src="https://github.com/user-attachments/assets/1ffd4dfd-c89f-42d9-ad26-9a39c5be835f"
/>


### How to test it?
* Start Kibana
* Create an event with `user.name: EC2AMAZ-R2R92FT\Administrator`
* Run the risk score engine
* Open the entity flyout

elasticmachine commented:
Pinging @elastic/response-ops (Team:ResponseOps)

davismcphee committed:
[Discover] Fix Discover session compatibility with deprecated saved objects API (#237557)

## Summary

This PR fixes compatibility between the Discover session saved object
and the deprecated saved objects API by switching the `tabs` property
back to optional, and implementing an on-read backfill step within the
Discover session content management service to populate `tabs` when it
does not exist.

Fixes #237703.

### Checklist

- [ ] Any text added follows [EUI's writing
guidelines](https://elastic.github.io/eui/#/guidelines/writing), uses
sentence case text and includes [i18n
support](https://github.com/elastic/kibana/blob/main/src/platform/packages/shared/kbn-i18n/README.md)
- [ ]
[Documentation](https://www.elastic.co/guide/en/kibana/master/development-documentation.html)
was added for features that require explanation or tutorials
- [x] [Unit or functional
tests](https://www.elastic.co/guide/en/kibana/master/development-tests.html)
were updated or added to match the most common scenarios
- [ ] If a plugin configuration key changed, check if it needs to be
allowlisted in the cloud and added to the [docker
list](https://github.com/elastic/kibana/blob/main/src/dev/build/tasks/os_packages/docker_generator/resources/base/bin/kibana-docker)
- [x] This was checked for breaking HTTP API changes, and any breaking
changes have been approved by the breaking-change committee. The
`release_note:breaking` label should be applied in these situations.
- [x] [Flaky Test
Runner](https://ci-stats.kibana.dev/trigger_flaky_test_runner/1) was
used on any tests changed
- [x] The PR description includes the appropriate Release Notes section,
and the correct `release_note:*` label is applied per the
[guidelines](https://www.elastic.co/guide/en/kibana/master/contributing.html#kibana-release-notes-process)
- [x] Review the [backport
guidelines](https://docs.google.com/document/d/1VyN5k91e5OVumlc0Gb9RPa3h1ewuPE705nRtioPiTvY/edit?usp=sharing)
and apply applicable `backport:*` labels.

---------

Co-authored-by: kibanamachine <42973632+kibanamachine@users.noreply.github.com>

seanrathier committed:
skip agentless mki (#237940)

## Summary

Skip agentless MKI test because of failure


https://buildkite.com/elastic/appex-qa-serverless-kibana-ftr-tests/builds/6975/steps/canvas?sid=0199c07a-7314-4d5c-9104-5ddb8a822364

sabarasaba committed:
[Console] Stricter checks for load_from query param (#237599)

kibanamachine committed:
[api-docs] 2025-10-08 Daily api_docs build (#237955)

Generated by
https://buildkite.com/elastic/kibana-api-docs-daily/builds/1219

shahzad31 committed:
[Synthetics] Disable max attempts for private locations sync task  (#237784)

## Summary

Disable max attempts for private locations sync task. Since task is a
scheduled task, if it fails, we don't need to retry, it will just retry
next time when it runs, in current state, if it retries, the multiple
task keeps doing the same work.


Also increased the timeout to 5 minutes and schedule to be less
aggressive to 10 minutes instead of 5 minutes.

kowalczyk-krzysztof committed:
Reduce unused URLs test flakiness (#237881)

## Summary

This PR removes redundant assertions and simplifies the test to reduce
the flakiness.

Fixes: https://github.com/elastic/kibana/issues/234775

adcoelho commented on x-pack/platform/plugins/shared/task_manager/server/task_scheduling.ts:
```diff
@@ -284,12 +284,23 @@ export class TaskScheduling {
+        if (taskInstance.schedule && taskInstance.schedule.interval) {
+          const result = await this.bulkUpdateSchedules([taskInstance.id], taskInstance.schedule);
+          if (result.errors.length) {
+            throw new Error(
```
Can't `bulkUpdateSchedule` throw `VERSION_CONFLICT_STATUS` again?

jughosta committed:
[Discover BS] Fix the alignment for the status icon (#237891)

## Summary

Before:
<img width="469" height="99" alt="Screenshot 2025-10-07 at 17 55 33"
src="https://github.com/user-attachments/assets/41895cdf-640a-4998-8e43-e4f62ade70f7"
/>

After:
<img width="549" height="96" alt="Screenshot 2025-10-07 at 18 00 38"
src="https://github.com/user-attachments/assets/ad913340-745d-4931-b99b-8b703b955855"
/>



### Checklist

- [x] The PR description includes the appropriate Release Notes section,
and the correct `release_note:*` label is applied per the
[guidelines](https://www.elastic.co/guide/en/kibana/master/contributing.html#kibana-release-notes-process)
- [x] Review the [backport
guidelines](https://docs.google.com/document/d/1VyN5k91e5OVumlc0Gb9RPa3h1ewuPE705nRtioPiTvY/edit?usp=sharing)
and apply applicable `backport:*` labels.

talboren committed:
[One Workflow] multiple tags view (#237562)

https://github.com/user-attachments/assets/9b6a8c9f-6e25-4c23-bb90-141c78d6cdbc

---------

Co-authored-by: Elastic Machine <elasticmachine@users.noreply.github.com>

pgayvallet committed:
[Agent Builder] improve NL-to-ESQL task (#237427)

## Summary

Fix https://github.com/elastic/search-team/issues/11454

This PR implements an agent builder specific version of the NL-to-ESQL
task, re-using parts of the "legacy" one (prompts provided by the ML
team, documentation, autocorrect logic) while using a new,
langgraph-driven, flow.

The main reasons behind the decision of coding our own version is simply
to introduce more control. The current NL-to-ESQL task is being used by
multiple consumers, which was making it very difficult to improve,
change, or enhance, without introducing risks, or just breaking things,
for other consumers (e.g current task is using streaming mode for o11y,
so we couldn't change it. o11y also has FTR tests asserting stuff
against the internals of the task, so anything we changed was breaking
those, and so on).

### Goodies

What does this new version bring:
- **in-task query execution**: Executing the ES|QL query is done within
the task, make it simpler for consumers, and also unlocking the next
point

- **retry-on-error**: After generating/validating the query, if the
execution returns an error, we loop back to the generation step,
providing the error to the LLM and asking it to try to fix it. We do
that `n` times (configurable, default to 3 generations - or 2 retries).

- **in-task resource / context description**: Resource / target
description (mappings, sample data and so on) generation is done from
within the task, so that consumers just have to pass the name of the
index/datastream/alias as a parameter.

- **cache-friendly**: The context we use for retries is append-only,
meaning that it is cache-able for providers having append-ish cache
methods.

- **maintainability and consistency**: Living in our codebase, using the
same tools and framework we use for our other tools and agentic flows.

- and last but not least, **freedom**: Of making it evolve however we
want, without being constrained by other consumers as we were for the
version from the inference plugin.

### Evals

`gpt-4o` evalutated by `gemini-2.5-pro`

We observe that the PR improves the metrics, especially for analytical
and hybrid queries (so basically where ES|QL is being used). Given the
actual prompts and logic didn't change significantly, it's likely that
those improvements are related to the "retry on error" logic, as a bad
query could previously cause the model to fail (especially with GPT-4
which tend to give us early in case of failures). The impact could be
less significant for Claude models (given they would try to call the
tool again more often), but even in that case, the latency should be
significantly improved given we avoid unnecessary roundtrips in case of
bad generation / invalid queries.

(Ran a couple other times to make sure, results are quite similar so
it's not just noise)

Branch

```
╔════════════════════════╤═════════════╤════════════╤═══════════════╤════════════╤═══════════════╗
║ Dataset                │  # Examples │ Factuality │  Groundedness │  Relevance │  Seq Accuracy ║
╟────────────────────────┼─────────────┼────────────┼───────────────┼────────────┼───────────────╢
║ text-retrieval-queries │      3 x 29 │      50.8% │         54.6% │      56.3% │         93.8% ║
╟────────────────────────┼─────────────┼────────────┼───────────────┼────────────┼───────────────╢
║ analytical-queries     │      3 x 15 │      45.7% │         74.8% │      84.0% │        100.0% ║
╟────────────────────────┼─────────────┼────────────┼───────────────┼────────────┼───────────────╢
║ hybrid-queries         │       3 x 9 │      15.0% │         69.0% │      71.2% │        100.0% ║
╟────────────────────────┼─────────────┼────────────┼───────────────┼────────────┼───────────────╢
║ unanswerable-queries   │      3 x 13 │      32.5% │         35.5% │      60.5% │         95.0% ║
╟────────────────────────┼─────────────┼────────────┼───────────────┼────────────┼───────────────╢
║ ambiguous-queries      │       3 x 4 │      42.9% │         24.9% │      67.6% │        100.0% ║
╟────────────────────────┼─────────────┼────────────┼───────────────┼────────────┼───────────────╢
║ Overall                │      3 x 70 │      41.3% │         55.5% │      65.6% │         96.5% ║
╚════════════════════════╧═════════════╧════════════╧═══════════════╧════════════╧═══════════════╝
```

Baseline

```
╔════════════════════════╤═════════════╤════════════╤═══════════════╤════════════╤═══════════════╗
║ Dataset                │  # Examples │ Factuality │  Groundedness │  Relevance │  Seq Accuracy ║
╟────────────────────────┼─────────────┼────────────┼───────────────┼────────────┼───────────────╢
║ text-retrieval-queries │      3 x 29 │      51.4% │         54.5% │      52.9% │         93.7% ║
╟────────────────────────┼─────────────┼────────────┼───────────────┼────────────┼───────────────╢
║ analytical-queries     │      3 x 15 │      44.7% │         79.2% │      87.4% │        100.0% ║
╟────────────────────────┼─────────────┼────────────┼───────────────┼────────────┼───────────────╢
║ hybrid-queries         │       3 x 9 │      10.8% │         47.0% │      69.6% │         94.1% ║
╟────────────────────────┼─────────────┼────────────┼───────────────┼────────────┼───────────────╢
║ unanswerable-queries   │      3 x 13 │      29.7% │         28.0% │      58.6% │         94.5% ║
╟────────────────────────┼─────────────┼────────────┼───────────────┼────────────┼───────────────╢
║ ambiguous-queries      │       3 x 4 │      38.8% │          8.3% │      52.0% │        100.0% ║
╟────────────────────────┼─────────────┼────────────┼───────────────┼────────────┼───────────────╢
║ Overall                │      3 x 70 │      40.0% │         51.3% │      63.4% │         95.6% ║
╚════════════════════════╧═════════════╧════════════╧═══════════════╧════════════╧═══════════════╝
```

Kerry350 committed:
[Streams] Fix delete step button enabled state (#237835)

## Summary

Closes https://github.com/elastic/kibana/issues/237453.

The button should not be enabled unless `manage` permissions are
available.

It would be nice to cascade some of these guards to the individual step
state machines too, but it means passing the definition down etc, so
this just uses the main state machine same as the other permission
checks.

(We may also just want to put this behind a `hasSimulatePrivileges`
guard, the same as many other events, which would let the step be
deleted and simulated, and the save button would still be behind the
`manage` permissions check).

delanni committed:
`yarn kbn clean` should also clean `.moon/cache` (#237967)

## Summary
As the title says.

stratoula committed:
[ES|QL] Supports solutions recommended queries for TS command (#237604)

## Summary

Part of https://github.com/elastic/kibana/issues/237425

Adds support of recommended queries in the TS command. TS command doesnt
support count(*) so I am only suggesting the solutions registered
queries.

<img width="840" height="209" alt="image"
src="https://github.com/user-attachments/assets/a8f53f1b-9a94-4e82-91a7-02874ade84ee"
/>

### Checklist

- [ ] [Unit or functional
tests](https://www.elastic.co/guide/en/kibana/master/development-tests.html)
were updated or added to match the most common scenarios

elastic-vault-github-plugin-prod[bot] committed:
[main] Sync bundled packages with Package Storage (#237755)

Automated by
https://buildkite.com/elastic/package-storage-infra-kibana-discover-release-branches/builds/3548

Co-authored-by: elastic-vault-github-plugin-prod <elasticmachine@elastic.co>
Co-authored-by: Julia Bardi <90178898+juliaElastic@users.noreply.github.com>

hop-dev committed:
[Entity Analytics][Risk Score] Use ESQL for calculating risk scores (#237871)

## Summary

Entity risk scoring now uses ESQL to calculate the scores, removing our
dependency on the deprecated scripted metric aggregation.


Changes:

- risk score now limited to 10 decimal places
- added global weights to esql risk scoring
- bug: after keys was adding an entity in esql risk scoring
- bug: category_1_score was being normalised
- bug: category_1_count was always 1
- bug: ESQL risk scoring did not work in non-default space

LucaWintergerst committed:
[Streams] prompt improvements, sanitize name (#237787)

The existing partitioning prompting was prone to call any Sub Streams
`logs.nginx-logs`, `logs.apache-logs`, often adding `-logs` to their
name.
This is because our prompts had lots of examples of this. 

We also did not do any name sanitization or validation prior, which
could cause problems when the LLM decided to name a Sub Stream
`logs.postgresDatabase` with an uppercase letter as one example.

hop-dev committed:
[Entity Analytics][Risk Score] Enable the risk score reset feature (#237829)

## Summary

This will mean that when users de-select "Retain previously calculated
risk scores" Entities which no longer have any alerts to be considered
for risk scoring will have their risk score set to 0 (instead of staying
at the previous score)

Also fixed a bug where scores were not being reset based on the saved
object setting, only the experimental feature.

Also makde it so the UI is hidden by the feature flag, just in case we
need to hide the feature for any reason,

<img width="717" height="582" alt="Screenshot 2025-10-07 at 13 20 18"
src="https://github.com/user-attachments/assets/5483d733-c150-472e-afeb-8609a4b446cc"
/>

elena-shostak committed:
[authn] Introduced error for empty login attempt (#237611)

## Summary

Issue is coming from the `value` being `undefined` in the
[attempt](https://github.com/elastic/kibana/blob/main/x-pack/platform/plugins/shared/security/server/routes/authentication/common.ts).
Added a separate error for empty login attempt with saml/oidc providers.


## How to test

Run `node scripts/functional_tests_server
--config=x-pack/platform/test/security_functional/oidc.config.ts`

and make a request
```bash
curl --location 'http://localhost:5620/internal/security/login' \
--header 'x-elastic-internal-origin: kibana' \
--header 'Kbn-Xsrf: true' \
--header 'Content-Type: text/plain' \
--data '{
    "providerType": "OIDC",
    "providerName": "oidc1",
    "currentURL": "http://localhost:5620/login?next=%2F"
}'
```

Refer to
https://github.com/elastic/kibana/issues/236579#issuecomment-3362049049
for debug details


### Checklist

- [x] The PR description includes the appropriate Release Notes section,
and the correct `release_note:*` label is applied per the
[guidelines](https://www.elastic.co/guide/en/kibana/master/contributing.html#kibana-release-notes-process)
- [x] Review the [backport
guidelines](https://docs.google.com/document/d/1VyN5k91e5OVumlc0Gb9RPa3h1ewuPE705nRtioPiTvY/edit?usp=sharing)
and apply applicable `backport:*` labels.

__Closes: https://github.com/elastic/kibana/issues/236579__

### Release Note
Introduced a separate error for empty login attempt with `saml/oidc`
providers.

maryam-saeidi committed:
[bkn/bench] Add --config-from-cwd cli option (#237501)

## Summary

Add a new CLI option to use cwd for config path instead of workspace
root.

MiriamAparicio committed:
[Metrics][Discover] Fix metric details flyout overlap (#237249)

Closes https://github.com/elastic/kibana/issues/236785

### Metric details flyout when opening Create alert flyout

#### Before


https://github.com/user-attachments/assets/d1d66563-79f8-46a4-a22a-948cb8cb4023

#### After


https://github.com/user-attachments/assets/3ae11be5-9268-4b06-a06e-b21698b5f6e0

### Metric details flyout when opening Inspect flyout

#### Before


https://github.com/user-attachments/assets/1a564d46-257e-49fe-bb62-b67eff4ec1cd

#### After


https://github.com/user-attachments/assets/58d3ca11-ab2f-49fa-a0d5-e15e6259a405

ymao1 committed:
PR feedback
